### ISSUE TITLE
Where is the VPK?

### ISSUE THREAD

**Issue by saildot4k** [SKGleba/yamt-vita#18]
No VPK is found to install.

**Comment by evertonstz** [comment]
#16

**Comment by RinMaru** [comment]
is there a tutorial on how to compile this?

**Comment by evertonstz** [comment]
> is there a tutorial on how to compile this?

I did it some months ago but my memory is kinda bad, I think you only need basic building tools on Linux like make and cmake but I ran into some problems and we discussed it at #15 
Recently SKG added [this script here](https://github.com/SKGleba/yamt-vita/blob/master/create_vpk.sh), I didn't test if it's working tho, but should be by the looks of it.

TL;DR: clone the repo and run create_vpk.sh, if it doesn't work try compiling everything manually like I did at #15

EDIT: I think  you'll also need [vitasdk](https://vitasdk.org/)

**Comment by RinMaru** [comment]
> > is there a tutorial on how to compile this?
> 
> I did it some months ago but my memory is kinda bad, I think you only need basic building tools on Linux like make and cmake but I ran into some problems and we discussed it at #15
> Recently SKG added [this script here](https://github.com/SKGleba/yamt-vita/blob/master/create_vpk.sh), I didn't test if it's working tho, but should be by the looks of it.
> 
> TL;DR: clone the repo and run create_vpk.sh, if it doesn't work try compiling everything manually like I did at #15
> 
> EDIT: I think you'll also need [vitasdk](https://vitasdk.org/)

and... im completly lost lol

**Comment by evertonstz** [comment]
> > > is there a tutorial on how to compile this?
> > 
> > 
> > I did it some months ago but my memory is kinda bad, I think you only need basic building tools on Linux like make and cmake but I ran into some problems and we discussed it at #15
> > Recently SKG added [this script here](https://github.com/SKGleba/yamt-vita/blob/master/create_vpk.sh), I didn't test if it's working tho, but should be by the looks of it.
> > TL;DR: clone the repo and run create_vpk.sh, if it doesn't work try compiling everything manually like I did at #15
> > EDIT: I think you'll also need [vitasdk](https://vitasdk.org/)
> 
> and... im completly lost lol

Hmm, have you ever compiled anything in C? There's literally no difference, you can find tons of tutorials online (but it's not even needed, all the hard part is done automatically by create_vpk.sh), the main burden here is setting vitasdk but that has nothing to do with yamt, you probably can ask for help in vitasdk's discord on how to set it up if you're having dificulties.  Once you have vitasdk set, everything should compile by running that create_vpk.sh file assuming you have cmake. Basically:

- install cmake
- install vitasdk
- clone this repository
- set the create_vpk.sh file as executable with `chmod +x create_vpk.sh`
- run create_vpk.sh

If the steps above sound alien to you you'll probably need to learn some linux basics before trying to compile this one. I would just get [yamt-lite](https://github.com/AuroraWright/yamt-vita-lite) at this point, Aurora releases vpks, but as far as I'm aware her base is 3.0, not 3.1.

**Comment by RinMaru** [comment]
like im told a portable version of git can do it but i have no clue how to set up vita sdk or start the compilation

**Comment by evertonstz** [comment]
> like im told a portable version of git can do it but i have no clue how to set up vita sdk or start the compilation

Wait, you're trying to do this on Windows? Forget about it, mate, you have to it on Linux or WSL. Git is not needed to compile, all you need is gcc, cmake and vitasdk. Like I said, without the basics of Linux this job is impossible, use yamt-lite for now.

**Comment by RinMaru** [comment]
> > like im told a portable version of git can do it but i have no clue how to set up vita sdk or start the compilation
> 
> Wait, you're trying to do this on Windows? Forget about it, mate, you have to it on Linux or WSL. Git is not needed to compile, all you need is gcc, cmake and vitasdk. Like I said, without the basics of Linux this job is impossible, use yamt-lite for now.

do you have a build you can send me then lol

**Comment by evertonstz** [comment]
> > > like im told a portable version of git can do it but i have no clue how to set up vita sdk or start the compilation
> > 
> > 
> > Wait, you're trying to do this on Windows? Forget about it, mate, you have to it on Linux or WSL. Git is not needed to compile, all you need is gcc, cmake and vitasdk. Like I said, without the basics of Linux this job is impossible, use yamt-lite for now.
> 
> do you have a build you can send me then lol

Nope, I didn't bother updating my vita from 3.0 to 3.1, I just compiled it for the fun of it back then. I also can't compile it for you because I'm also on Windows currently :/

**Comment by RinMaru** [comment]
lol ok thanks anyways

**Comment by SKGleba** [comment]
The next release will have a precompiled binary.

**Comment by miinht** [comment]
Does a precompiled binary mean a vpk will come included? Or do we still have to compile it ourselfs manually on Linux? Also what new features/changes will be in the next release?

**Comment by evertonstz** [comment]
> Does a precompiled binary mean a vpk will come included? Or do we still have to compile it ourselfs manually on Linux? Also what new features/changes will be in the next release?

Yes, precompiled binary is the vpk. You probably can track the new changes here once they're commited: https://github.com/SKGleba/yamt-vita/commits/master

**Comment by miinht** [comment]
Ah ok, here's the link to the vpk for v3.1 if anyone is still looking. Probably unnecessary since next release will come with the vpk anyways.

https://drive.google.com/file/d/1nLa8aFo_z6wp3LgybPz7gb3cCl4OQc51/view
